Applications/opensense/opensense.cpp

```diff
@@ -231,7 +231,7 @@ TimeSeriesTable_<SimTK::Quaternion> readRotationsFromXSensFiles(const std::strin
 {
     XsensDataReaderSettings readerSettings(readerSetupFile);
     XsensDataReader reader(readerSettings);
-    DataAdapter::OutputTables tables = reader.extendRead(directory);
```

Comment:
I think the initial design was OK but incomplete. The base `DataAdapter` should provide the public interface `read(const string& sourceIdenitifier)` and delegate to the virtual method implemented by concrete readers. That design enables us to put common operations (checks) in the base `read` method while delegating details to each individual reader. I would not simply rename `extendRead` here.